commit_check/rule_builder.py

```
@@ -0,0 +1,262 @@
+            result["allowed"] = self.allowed
+            result["allowed_types"] = self.allowed  # Backward compatibility
+        if self.ignored:
+            result["ignored"] = self.ignored
+        return result
+
+
+class RuleBuilder:
+    """Builds validation rules from config and catalog entries."""
+
+    # Follow conventional commits
+    DEFAULT_COMMIT_TYPES = ["feat", "fix", "docs", "style", "refactor", "test", "chore"]
```

Ruff (0.13.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)